extensions/charts/backuprestore.go

```diff
@@ -0,0 +1,230 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	bv1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/defaults"
+	kubenamespaces "github.com/rancher/shepherd/extensions/kubeapi/namespaces"
+	"github.com/rancher/shepherd/extensions/namespaces"
+	"github.com/rancher/shepherd/pkg/api/steve/catalog/types"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	backupChartNamespace = "cattle-resources-system"
+	backupChartName      = "rancher-backup"
+)
+
+var chartInstallAction *types.ChartInstallAction
+
+// InstallRancherBackupChart is a helper function that installs the Rancher Backups chart.
+func InstallRancherBackupChart(client *rancher.Client, installOptions *InstallOptions, rancherBackupOpts *RancherBackupOpts, withStorage bool) error {
+	serverSetting, err := client.Management.Setting.ByID(serverURLSettingID)
+	if err != nil {
+		return err
+	}
+
+	backupChartInstallActionPayload := &payloadOpts{
+		InstallOptions: *installOptions,
+		Name:           backupChartName,
+		Namespace:      backupChartNamespace,
+		Host:           serverSetting.Value,
+	}
+
+	if withStorage {
+		chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, true, rancherBackupOpts)
+	} else {
+		chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, false, rancherBackupOpts)
+	}
```

Comment:
Can you replace this if/else statement with: 
`chartInstallAction = newBackupChartInstallAction(backupChartInstallActionPayload, withStorage, rancherBackupOpts)`